muqa-org/zazelenimo
# feature: add round selector

## 🎯 Overview
This PR introduces round selection functionality and improves application data handling throughout the platform. It includes significant updates to the project discovery flow and adds support for OP Sepolia network.

## 🔍 Key Changes

### Round Selection & Context
- Added global round ID context for consistent round selection across components
- Added ability to dynamically change the round if `NEXT_PUBLIC_ALLOW_ROUND_INPUT` is true
- Moved round selector to footer 
- Implemented real application data loading with fallback to dummy data

### Project Discovery Improvements
- Moved projects to root dir (no longer under `[roundId]`)
- Hooked neighborhood filtering functionality
- Implemented "No applications" message for empty states
- Temporarily disabled pagination for initial release
- Added loading states for application data
- Fixed footer positioning on empty states

### Network Support
- Added OP Sepolia environment support
- Updated chain configuration
- Replaced chain data with forked version for Optimism Sepolia and EURMQC support

### UI/UX Enhancements
- Added cart change animations
- Improved euro amount rendering
- Fixed details rerendering on donation changes
- Improved footer rendering across screen sizes

### Chores
- updated the README

## 🧪 Testing
1. Switch between different rounds using the round selector
2. Verify application loading states
3. Test neighborhood filtering
4. Confirm OP Sepolia network connectivity
5. Verify cart animations and euro rendering

## 📝 Notes
- Dummy data support can be toggled via `NEXT_PUBLIC_USE_DUMMY_DATA` env var
- Custom IPFS URL added for Optimism Sepolia support
- Custom indexer URL added for Optimism Sepolia support
- Some type issues resolved for better TypeScript compliance

## 📦 Dependencies
- Updated chain data package to forked version
- Added animation dependencies for cart changes

## 🚀 Deployment Notes
- Requires new environment variables:
  - `NEXT_PUBLIC_USE_DUMMY_DATA` (optional)
  - `NEXT_PUBLIC_IPFS_GATEWAY_URL` (optional)
  - `NEXT_PUBLIC_INDEXER_URL` (optional)
